assets/js/googlesitekit/modules/datastore/settings.test.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * `googlesitekit/modules` datastore: settings tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import Modules from 'googlesitekit-modules';
+import { STORE_NAME } from './constants';
+import { createTestRegistry } from '../../../../../tests/js/utils';
+
+describe( 'core/modules store changes', () => {
+	let registry;
+	const slug = 'test-module';
+	const nonExistentModuleSlug = 'not-module';
+	const moduleStoreName = `modules/${ slug }`;
+	let moduleCanSubmitChanges = false;
+	let submittingChanges = false;
+
+	beforeAll( () => {
+	} );
+
+	beforeEach( () => {
+		const storeDefinition = Modules.createModuleStore( moduleStoreName );
+		const submitChanges = () => {
+			return {
+				payload: {},
+				type: 'DUMMY_ACTION',
+			};
+		};
+		const canSubmitChanges = () => moduleCanSubmitChanges;
+		const isDoingSubmitChanges = () => {
+			return submittingChanges;
+		};
+
+		registry = createTestRegistry();
+		registry.registerStore( moduleStoreName, Data.combineStores(
+			storeDefinition,
+			{
+				actions: {
+					submitChanges,
+				},
+				selectors: {
+					canSubmitChanges,
+					isDoingSubmitChanges,
+				},
+			}
+		) );
+	} );
+
+	afterAll( () => {
+	} );
+
+	afterEach( () => {
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'submitting changes', () => {
+			it( 'is submitting changes', async () => {
+				expect( registry.select( STORE_NAME ).isDoingSubmitChanges( nonExistentModuleSlug ) ).toBe( false );
+
+				expect( registry.select( STORE_NAME ).isDoingSubmitChanges( slug ) ).toBe( false );
+				submittingChanges = true;
+				expect( registry.select( STORE_NAME ).isDoingSubmitChanges( slug ) ).toBe( true );
+			} );
+
+			it( 'can submit changes', () => {
```

Comment:
Regarding the test names, I think we can word it a little differently to better describe what the actions and selectors do. This applies to all of the cases here, but something like "it proxies the selector call to the module with the given slug" would be a bit more descriptive and useful for someone who didn't know what it does. Even if the name for this test might be the same for each selector/action, it would still be scoped per-selector. Because of the structure of the tests, each test is already in the context of a specific selector/action so it's best to use the name to explain more about what's happening. Again, there are lots of examples to reference from other stores, but here's a bit more of a [comprehensive example](https://github.com/google/site-kit-wp/blob/6beb2381961962da4cfeedde0a2d027e95128662/assets/js/modules/tagmanager/datastore/versions.test.js) from one of the partials that I worked on recently.

Reply:
Addressed.